### PR TITLE
cache: render area labels on the map

### PR DESCRIPTION
**Pull request by christoabrown**
Parses the map label locations and fonts from the cache and updates MapImageDumper to render the labels.
  
Screenshot 2024-04-07 151640

### CODE REVIEW

File: cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapDataLoader.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	{
+		int worldMapDataType = in.readUnsignedByte();
+
+		if (worldMapDataType != 0)
+		{
+			throw new RuntimeException("Expected worldMapDataType 0 got " + worldMapDataType);
+		}
+
+		WorldMapData0Definition worldMapData0Definition = new WorldMapData0Definition();
+		worldMapData0Definition.field1 = in.readUnsignedByte();
```
Comment (Adam-) [inline cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapDataLoader.java:43]: names from Polar:
```java
            this.minLevel = var1.g1();
            this.levels = var1.g1();
            this.displaySquareX = var1.g2();
            this.displaySquareZ = var1.g2();
            this.sourceSquareX = var1.g2();
            this.sourceSquareZ = var1.g2();
            this.groupId = var1.gSmart2or4null();
            this.fileId = var1.gSmart2or4null();
```

File: cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapDataLoader.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	{
+		int worldMapDataType = in.readUnsignedByte();
+
+		if (worldMapDataType != 1)
+		{
+			throw new RuntimeException("Expected worldMapDataType 1 got " + worldMapDataType);
+		}
+
+		WorldMapData1Definition worldMapData1Definition = new WorldMapData1Definition();
+		worldMapData1Definition.field1 = in.readUnsignedByte();
```
Comment (Adam-) [inline cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapDataLoader.java:65]: more names from Polar:
```java
            this.minLevel = var1.g1();
            this.levels = var1.g1();
            this.displaySquareX = var1.g2();
            this.displaySquareZ = var1.g2();
            this.displayZoneX = var1.g1();
            this.displayZoneY = var1.g1();
            this.sourceSquareX = var1.g2();
            this.sourceSquareZ = var1.g2();
            this.sourceZoneX = var1.g1();
            this.sourceZoneZ = var1.g1();
            this.groupId = var1.gSmart2or4null();
            this.fileId = var1.gSmart2or4null();
```

File: cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapCompositeLoader.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		WorldMapDataLoader worldMapDataLoader = new WorldMapDataLoader();
+		WorldMapElementLoader worldMapElementLoader = new WorldMapElementLoader();
+
+		WorldMapCompositeDefinition worldMapCompositeDefinition = new WorldMapCompositeDefinition();
+		InputStream in = new InputStream(buffer);
+
+		int worldData0Count = in.readUnsignedShort();
+		for (int i = 0; i < worldData0Count; ++i)
+		{
+			worldMapCompositeDefinition.worldMapData0Definitions.add(worldMapDataLoader.load0(in));
```
Comment (Adam-) [inline cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapCompositeLoader.java:43]: type 0 is `mapsquare`

File: cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapCompositeLoader.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		}
+
+		int worldData1Amount = in.readUnsignedShort();
+		for (int i = 0; i < worldData1Amount; ++i)
+		{
+			worldMapCompositeDefinition.worldMapData1Definitions.add(worldMapDataLoader.load1(in));
```
Comment (Adam-) [inline cache/src/main/java/net/runelite/cache/definitions/loaders/WorldMapCompositeLoader.java:49]: type 1 is `zone`

**Comment by christoabrown** [comment]
@Adam- Renamed those files and updated the field names
